NebulaWorld/Combat/CombatManager.cs

```diff
@@ -0,0 +1,263 @@
+            Players[0].position = GameMain.mainPlayer.position;
+            Players[0].uPosition = GameMain.mainPlayer.uPosition;
+            Players[0].isAlive = GameMain.mainPlayer.isAlive;
+            var macha = GameMain.mainPlayer.mecha;
+            Players[0].mecha = macha;
+            Players[0].skillTargetL = macha.skillTargetLCenter;
+            Players[0].skillTargetULast = Players[0].skillTargetU;
+            Players[0].skillTargetU = macha.skillTargetUCenter;
+
+            ActivedPlanets.Add(Players[0].planetId);
+            ActivedStars.Add(Players[0].starId);
+            if (Players[0].planetId <= 0 && Players[0].starId > 0)
+            {
+                ActivedStarsMechaInSpace.Add(Players[0].starId);
+            }
+            IndexByPlayerId[Players[0].id] = 0;
+
+            var localPlanetId = Players[0].planetId;
+            var index = 1;
+            foreach (var pair in remotePlayersModels)
+            {
+                var snapshot = pair.Value.Movement.GetLastPosition();
+                var planetData = GameMain.galaxy.PlanetById(snapshot.LocalPlanetId);
+                var player = pair.Value.PlayerInstance;
+                if (planetData != null) // On planet
+                {
+                    player.uPosition = planetData.uPosition + (VectorLF3)(planetData.runtimeRotation * player.position);
+                    player.uRotation = planetData.runtimeRotation * Maths.SphericalRotation(player.position, 0f);
+                }
+                else // In space
+                {
+                    player.uPosition = pair.Value.Movement.absolutePosition;
+                }
+                ref var ptr = ref Players[index];
+                ptr.id = pair.Key;
+                ptr.planetId = snapshot.LocalPlanetId;
+                ptr.starId = pair.Value.Movement.LocalStarId;
+                // If the remote player is on the same planet, player.position is more precise
+                // Otherwise it has to use the interpolated recevied position
+                ptr.position = ptr.id == localPlanetId ? player.position : snapshot.LocalPlanetPosition.ToVector3();
+                ptr.uPosition = player.uPosition;
+                ptr.isAlive = player.isAlive;
+
+                macha = pair.Value.MechaInstance;
+                ptr.mecha = macha;
+                ptr.skillTargetL = macha.skillTargetLCenter;
+                ptr.skillTargetULast = ptr.skillTargetU;
+                ptr.skillTargetU = macha.skillTargetUCenter;
+                macha.energyShieldEnergy = macha.energyShieldEnergyRate > 1 ? 0 : int.MaxValue;
+
+                ActivedPlanets.Add(ptr.planetId);
+                ActivedPlanets.Add(ptr.starId);
+                if (ptr.planetId <= 0 && ptr.starId > 0)
+                {
+                    ActivedStarsMechaInSpace.Add(ptr.starId);
+                }
+                IndexByPlayerId[pair.Key] = index;
+                ++index;
```

Comment:
Should this not be `index++;`? (if I remember correctly `++index;` does not actually increment the variable and only makes sense if you are assigning this to another variable or using this in a calculation of some sort)

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#1296-prefix-increment-and-decrement-operators
> The resulting value is stored in the location given by the evaluation of x.

Reply:
```cs
IndexByPlayerId[pair.Key] = index;
++index;
```
could be
``
IndexByPlayerId[pair.Key] = index++;
```
though, no?

Reply:
yeah, I can change to this.